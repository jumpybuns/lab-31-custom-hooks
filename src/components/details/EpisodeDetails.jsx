import React from 'react';
import PropTypes from 'prop-types';

function EpisodeDetails({ name }) {
  return (
    <figure>
      <h2>{name}</h2>
    </figure>
  );
}

EpisodeDetails.propTypes = {
  name: PropTypes.string.isRequired,
};

export default EpisodeDetails;
