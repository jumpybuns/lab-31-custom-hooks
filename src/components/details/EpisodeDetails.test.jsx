import React from 'react';
import { render, cleanup } from '@testing-library/react';
import EpisodeDetails from './EpisodeDetails';

describe('EpisodeDetails component', () => {
  afterEach(() => cleanup());
  it('renders EpisodeDetails', () => {
    const { asFragment } = render(
      <EpisodeDetails
        episodes={{
          id: 2,
          name: 'Lawnmower Dog',
          air_date: 'December 9, 2013',
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
