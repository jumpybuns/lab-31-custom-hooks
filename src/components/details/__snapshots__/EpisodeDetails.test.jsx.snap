// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetails component renders EpisodeDetails 1`] = `
<DocumentFragment>
  <figure
    data-testid="episode"
  >
    <h2 />
    <h3 />
  </figure>
</DocumentFragment>
`;
